diff --git a/kaslr_offset/kernel_module/Makefile b/kaslr_offset/kernel_module/Makefile
index fa7c0bd..59c6dfa 100644
--- a/kaslr_offset/kernel_module/Makefile
+++ b/kaslr_offset/kernel_module/Makefile
@@ -1,8 +1,10 @@
 obj-m += direct_physical_map.o 
 
+KVERSION := $(shell uname -r)
+
 all:
-	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
+	make -C /lib/modules/$(KVERSION)/build M=$(PWD) modules
 
 clean:
-	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
+	make -C /lib/modules/$(KVERSION)/build M=$(PWD) clean
 
