diff --git a/Makefile b/Makefile
index 5c3dbb7..b469230 100644
--- a/Makefile
+++ b/Makefile
@@ -16,11 +16,11 @@ install:
 	install -d -m 755 $(DESTDIR)/usr/lib/mana-toolkit/
 	install -d -m 755 $(DESTDIR)/var/lib/mana-toolkit/sslsplit
 	install -d -m 755 $(DESTDIR)/etc/mana-toolkit/
-	install -d -m 755 $(DESTDIR)/etc/apache2/sites-available/
+	install -d -m 755 $(DESTDIR)/etc/httpd/conf/extra
 	# Install configuration files
 	install -m 644 run-mana/conf/* $(DESTDIR)/etc/mana-toolkit/
 	install -m 644 crackapd/crackapd.conf $(DESTDIR)/etc/mana-toolkit/
-	install -m 644 apache/etc/apache2/sites-available/* $(DESTDIR)/etc/apache2/sites-available/
+	install -m 644 apache/etc/apache2/sites-available/* $(DESTDIR)/etc/httpd/conf/extra
 	# Install the hostapd binary
 	install -m 755 hostapd-mana/hostapd/hostapd $(DESTDIR)/usr/lib/mana-toolkit/
 	install -m 755 hostapd-mana/hostapd/hostapd_cli $(DESTDIR)/usr/lib/mana-toolkit/
