diff --git a/setup.py b/setup.py
index a07184d..3b8f908 100644
--- a/setup.py
+++ b/setup.py
@@ -26,7 +26,6 @@ setup(name='pygraphml',
       author='Hadrien Mary',
       author_email='hadrien.mary@gmail.com',
       description='Small library to parse GraphML files in Python',
-      long_description=read('README.md'),
       url='https://github.com/hadim/pygraphml/',
       packages=['pygraphml'],
       classifiers=[
