diff --git a/guymager.pro b/guymager.pro
index 61091bb..fc17606 100644
--- a/guymager.pro
+++ b/guymager.pro
@@ -147,9 +147,6 @@ HEADERS += threadscan.h
 HEADERS += threadwrite.h
 HEADERS += util.h
 
-QMAKE_CXXFLAGS *= $$system(dpkg-buildflags --get CXXFLAGS)
-QMAKE_LFLAGS   *= $$system(dpkg-buildflags --get LDFLAGS)
-
 QMAKE_CXXFLAGS         += -fmessage-length=0     # Tell g++ not to split messages into different lines
 QMAKE_CXXFLAGS         += -fno-strict-aliasing   # Avoid strange error messages when using QVarLengthArray
 
@@ -169,7 +166,7 @@ LIBS += -lz
 LIBS += -ldl
 
 
-!contains( DEFINES, "ENABLE_LIBEWF=0") {   # Corresponds to logic in common.h: ENABLE_LIBEWF must be set explicetly to 0 or else it is 1 (i.e. if it's not set or set to something different than 0)
+!contains(DEFINES, "ENABLE_LIBEWF=0") {
    USE_LIBEWF=1
 } else {
    USE_LIBEWF=0
@@ -199,12 +196,12 @@ equals(STATIC, "yes") {
       }
    }
 }
-   LIBS += -lguytools
-   contains (USE_LIBEWF, 1) {
-      message ("Using libewf!")
-      LIBS += -lewf
-      LIBS += -lbfio
-   }
+
+LIBS += -lguytools
+contains (USE_LIBEWF, 1) {
+   message ("Using libewf!")
+   LIBS += -lewf
+   LIBS += -lbfio
 }
 
 contains (USE_LIBEWF, 0) {
