--- build.gradle	2021-02-15 15:22:05.000000000 -0500
+++ ../../build.gradle	2021-03-12 11:20:40.107098281 -0500
@@ -192,9 +192,6 @@
     ant.exec(executable: "chmod", dir: dist, failonerror: true) {
         ant.arg(line: "a+x usr/bin/ipscan usr/lib/ipscan/${project.name}-${platform}-${version}.jar")
     }
-    ant.exec(executable: 'fakeroot', dir: dist + '/..', failonerror: true) {
-        ant.arg(line: "dpkg-deb -b deb ${project.name}_${version}_${arch}.deb")
-    }
     ant.delete(dir: dist)
 }
 
@@ -209,10 +206,6 @@
         ant.replacefilter(token: "RPM_VERSION", value: rpmVersion)
         ant.replacefilter(token: "VERSION", value: version)
     }
-    ant.exec(executable: "rpmbuild", dir: dist, failonerror: true) {
-        ant.arg(line: "--target ${arch} --define \"_topdir ${new File(dist).absolutePath}\" --define \"platform ${platform}\" -bb SPECS/ipscan.spec")
-    }
-    ant.move(file: "${dist}/RPMS/${arch}/ipscan-${rpmVersion}-1.${arch}.rpm", todir:'build/libs')
     ant.delete(dir: dist)
 }
 
