diff -aur halberd-0.2.4.orig/setup.py halberd-0.2.4/setup.py
--- halberd-0.2.4.orig/setup.py	2016-05-29 19:50:05.769317402 -0400
+++ halberd-0.2.4/setup.py	2016-05-29 19:50:31.555615574 -0400
@@ -108,8 +108,8 @@
     packages = ['Halberd', 'Halberd.clues'],
     package_dir = {'Halberd': 'Halberd'},
     scripts = [D('scripts', 'halberd')],
-    data_files = [(D('man', 'man1'), \
-                  [D('man', 'man1', 'halberd.1')])],
+    #data_files = [(D('man', 'man1'), \
+    #              [D('man', 'man1', 'halberd.1')])],
     classifiers = classifiers.splitlines(),
     cmdclass = {'test': test},
 )
