diff -aur capstone-3.0.3.orig/bindings/python/Makefile capstone-3.0.3/bindings/python/Makefile
--- capstone-3.0.3.orig/bindings/python/Makefile	2015-06-21 18:40:18.174045628 -0400
+++ capstone-3.0.3/bindings/python/Makefile	2015-06-21 18:50:04.255030217 -0400
@@ -14,7 +14,7 @@
 	rm -rf prebuilt/win64/capstone.dll
 	rm -rf prebuilt/win32/capstone.dll
 	if test -n "${DESTDIR}"; then \
-		python setup.py build -b $(OBJDIR) install --root="${DESTDIR}"; \
+		python2 setup.py build -b $(OBJDIR) install --root="${DESTDIR}" --optimize=1; \
 	else \
 		python setup.py build -b $(OBJDIR) install; \
 	fi
@@ -24,7 +24,7 @@
 	rm -rf prebuilt/win64/capstone.dll
 	rm -rf prebuilt/win32/capstone.dll
 	if test -n "${DESTDIR}"; then \
-		python3 setup.py build -b $(OBJDIR) install --root="${DESTDIR}"; \
+		python3 setup.py build -b $(OBJDIR) install --root="${DESTDIR}" --optimize=1; \
 	else \
 		python3 setup.py build -b $(OBJDIR) install; \
 	fi
