diff -aur pymssql-2.1.3.orig/_mssql.pyx pymssql-2.1.3/_mssql.pyx
--- pymssql-2.1.3.orig/_mssql.pyx	2016-06-23 11:05:23.282731666 -0400
+++ pymssql-2.1.3/_mssql.pyx	2016-06-23 11:08:18.648233780 -0400
@@ -1695,12 +1695,10 @@
         return DBVERSION_42
     if verstr == u'7.0':
         return DBVERSION_70
-    if verstr == u'7.1':
+    if verstr in (u'7.1', u'8.0'):
         return DBVERSION_71
     if verstr == u'7.2':
         return DBVERSION_72
-    if verstr == u'8.0':
-        return DBVERSION_80
     raise MSSQLException('unrecognized tds version: %s' % verstr)
 
 #######################
diff -aur pymssql-2.1.3.orig/sqlfront.pxd pymssql-2.1.3/sqlfront.pxd
--- pymssql-2.1.3.orig/sqlfront.pxd	2016-06-23 11:05:23.282731666 -0400
+++ pymssql-2.1.3/sqlfront.pxd	2016-06-23 11:08:45.491383635 -0400
@@ -106,7 +106,6 @@
     int DBVERSION_70
     int DBVERSION_71
     int DBVERSION_72
-    int DBVERSION_80
 
     ## Type Constants ##
     cdef enum:
