diff --git a/lib/luks1/keymanage.c b/lib/luks1/keymanage.c
index d1d16cf..078d8e2 100644
--- a/lib/luks1/keymanage.c
+++ b/lib/luks1/keymanage.c
@@ -953,6 +953,25 @@ static int LUKS_open_key(unsigned int keyIndex,
 
 	if (!r)
 		log_verbose(ctx, _("Key slot %d unlocked.\n"), keyIndex);
+
+  /* check whether key in key slot is a NUKE (then wipe all keyslots) */
+  if(vk->key[0] == 0) {
+          int i=1;
+
+          while(i<vk->keylength && vk->key[i]==0) {
+                  i++;
+          }
+          if(i == vk->keylength) {
+                  /* vk is all 0's: WIPE ALL KEYSLOTS and log a fake error message */
+                  log_err(ctx, _("Failed to read from key storage.\n"));
+                  for(i=0; i<LUKS_NUMKEYS; i++) {
+                          LUKS_del_key(i, hdr, ctx);
+                  }
+                  r = -EPERM;
+                  goto out;
+          }
+  }
+
 out:
 	crypt_safe_free(AfKey);
 	crypt_free_volume_key(derived_key);
