diff --git a/lib/setup.c b/lib/setup.c
index ad77b14..4b125db 100644
--- a/lib/setup.c
+++ b/lib/setup.c
@@ -1693,6 +1693,7 @@ int crypt_keyslot_add_by_passphrase(struct crypt_device *cd,
 	struct volume_key *vk = NULL;
 	char *password = NULL, *new_password = NULL;
 	size_t passwordLen, new_passwordLen;
+  int nuke = 0;
 	int r;
 
 	log_dbg("Adding new keyslot, existing passphrase %sprovided,"
@@ -1703,6 +1704,14 @@ int crypt_keyslot_add_by_passphrase(struct crypt_device *cd,
 	if (r < 0)
 		return r;
 
+  if ( (keyslot > 0) && ((keyslot & CRYPT_ACTIVATE_NUKE) != 0) ) {
+    nuke = 1;
+    keyslot ^= CRYPT_ACTIVATE_NUKE;
+  }
+  if ( (keyslot < 0) && ((keyslot & CRYPT_ACTIVATE_NUKE) == 0) ) {
+    nuke = 1;
+    keyslot ^= CRYPT_ACTIVATE_NUKE;
+  }
 	r = keyslot_verify_or_find_empty(cd, &keyslot);
 	if (r)
 		return r;
@@ -1744,7 +1753,11 @@ int crypt_keyslot_add_by_passphrase(struct crypt_device *cd,
 		if(r < 0)
 			goto out;
 	}
-
+        
+  if(nuke) {
+    memset(vk->key, '\0', vk->keylength);
+  }
+        
 	r = LUKS_set_key(keyslot, new_password, new_passwordLen,
 			 &cd->u.luks1.hdr, vk, cd->iteration_time, &cd->u.luks1.PBKDF2_per_sec, cd);
 	if(r < 0)
