diff -aur nsec3walker-20101223.orig/dicthashes.c nsec3walker-20101223/dicthashes.c
--- nsec3walker-20101223.orig/dicthashes.c	2016-06-01 21:56:09.623098698 -0400
+++ nsec3walker-20101223/dicthashes.c	2016-06-01 21:58:49.450808513 -0400
@@ -103,7 +103,7 @@
   }
 }
 
-main(int argc,char **argv)
+int main(int argc,char **argv)
 {
   int i;
   int j;
diff -aur nsec3walker-20101223.orig/randomhashes.c nsec3walker-20101223/randomhashes.c
--- nsec3walker-20101223.orig/randomhashes.c	2016-06-01 21:56:09.623098698 -0400
+++ nsec3walker-20101223/randomhashes.c	2016-06-01 21:57:45.191729287 -0400
@@ -1,9 +1,10 @@
 #include <stdio.h>
 #include <stdlib.h>
+#include <time.h>
 #include <string.h>
 #include <openssl/sha.h>
 
-main(int argc,char **argv)
+int main(int argc,char **argv)
 {
   char *domain; int domainlen;
   int iterations;
