diff --git a/profiles/general/default.cfg b/profiles/general/default.cfg
index d96ca0f..d78e81c 100644
--- a/profiles/general/default.cfg
+++ b/profiles/general/default.cfg
@@ -57,12 +57,12 @@ TOOL_THEHARVESTER_DIR: /usr/share/theharvester
 TOOL_METAGOOFIL_DIR: /usr/share/metagoofil
 # Backtrack 5 R1, no longer working in Backtrack 5 R2:
 #TOOL_HTTPRINT_DIR: /pentest/enumeration/www/httprint/linux
-TOOL_HTTPRINT_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/httprint/httprint_301/linux
+TOOL_HTTPRINT_DIR: /usr/share/httprint 
 TOOL_WAFW00F: /usr/bin/wafw00f
 TOOL_NMAP: /usr/bin/nmap
 TOOL_WHATWEB: /usr/bin/whatweb
 #TOOL_WHATWEB: @@@FRAMEWORK_DIR@@@/tools/restricted/whatweb/whatweb-0.4.7/whatweb
-TOOL_CMS_EXPLORER_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/cms-explorer/cms-explorer-1.0
+TOOL_CMS_EXPLORER_DIR: /usr/share/cms-explorer
 TOOL_LBD: /usr/bin/lbd
 #TOOL_REV_HOSTS_DIR: /pentest/enumeration/web/revhosts
 TOOL_DNS_RECON: /usr/bin/dnsrecon
@@ -84,18 +84,18 @@ TOOL_WAPITI_BIN: /usr/bin/wapiti
 # BackTrack:
 #TOOL_DIRBUSTER_BIN: DirBuster-0.12.jar
 # Kali Linux:
-TOOL_DIRBUSTER_BIN: /usr/share/dirbuster/DirBuster-1.0-RC1.jar
+TOOL_DIRBUSTER_BIN: /usr/bin/dirbuster
 #TOOL_DIRBUSTER_DIR: /usr/share/dirbuster
-TOOL_METASPLOIT_DIR: /usr/share/metasploit-framework
+TOOL_METASPLOIT_DIR: /usr/share/metasploit
 TOOL_HOPPY_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/hoppy-1.8.1/hoppy-1.8.1
 TOOL_WPSCAN: /usr/bin/wpscan
 TOOL_HYDRA: /usr/bin/hydra
 TOOL_TLSSLED: /usr/bin/tlssled
 TOOL_SSL_CIPHER_CHECK: @@@FRAMEWORK_DIR@@@/tools/restricted/ssl/ssl-cipher-check/ssl-cipher-check.pl
 TOOL_HTTP_TRACEROUTE: @@@FRAMEWORK_DIR@@@/tools/discovery/web/rev_proxy/HTTP-Traceroute.py
-TOOL_HTTP_DOS_SLOWLORIS: @@@FRAMEWORK_DIR@@@/tools/dos/web/slowloris.pl
+TOOL_HTTP_DOS_SLOWLORIS: /usr/bin/slowloris
 TOOL_HTTP_DOS_HASH_COLLISION: @@@FRAMEWORK_DIR@@@/tools/dos/web/HashCollision-DOS-POC/HashtablePOC.py
-TOOL_PANOPTIC_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/Panoptic
+TOOL_PANOPTIC_DIR: /usr/share/panoptic
 TOOL_DNSPIDER: @@@FRAMEWORK_DIR@@@/tools/restricted/dnspider/dnsspider-0.6.py
 TOOL_DNSPIDER_DIR: @@@FRAMEWORK_DIR@@@/tools/restricted/dnspider/
 TOOL_O-SAFT_BIN: /usr/bin/o-saft
