diff --git a/owtf.py b/owtf.py
index 21b7139..3e46b61 100755
--- a/owtf.py
+++ b/owtf.py
@@ -10,14 +10,12 @@ import os
 import sys
 import logging
 
-from framework.dependency_check import verify_dependencies
 from framework.core import Core
 from framework.dependency_management.component_initialiser import ComponentInitialiser, DatabaseNotRunningException
 from framework.dependency_management.dependency_resolver import ServiceLocator
 from framework import update
 from framework.lib.cli_options import usage, parse_options, parse_update_options
 
-verify_dependencies(os.path.dirname(os.path.abspath(sys.argv[0])) or '.')
 
 
 def banner():
