From 87d78ec488bfbbab3683b64c1c58b53b50c08803 Mon Sep 17 00:00:00 2001
From: Kubuxu <kubuxu@gmail.com>
Date: Tue, 1 Dec 2015 17:14:27 +0100
Subject: [PATCH] SECCOMP: bind and getsockaddr do not exist Introduced in
 3712d583b8ac7b940521d96107cb06

---
 util/Seccomp.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/util/Seccomp.c b/util/Seccomp.c
index 9cd01b3..4a2065e 100644
--- a/util/Seccomp.c
+++ b/util/Seccomp.c
@@ -314,9 +314,13 @@ static struct sock_fprog* mkFilter(struct Allocator* alloc, struct Except* eh)
         // Security_nofiles
         IFEQ(__NR_setrlimit, success),
 
-        // for ETHInterface_listDevices (netlink)
+        // for ETHInterface_listDevices (netlinkk)
+        #ifdef __NR_bind
         IFEQ(__NR_bind, bind_netlink),
+        #endif
+        #ifdef __NR_getsockname
         IFEQ(__NR_getsockname, success),
+        #endif
         RET(SECCOMP_RET_TRAP),
 
         LABEL(socket),
