diff --git a/archinfo/arch.py b/archinfo/arch.py
index 59f8b1e..70caea2 100644
--- a/archinfo/arch.py
+++ b/archinfo/arch.py
@@ -211,6 +211,8 @@ class Arch:
         # generate register mapping (offset, size): name
         self.register_size_names = {}
         for reg in self.register_list:
+            if reg.vex_offset is None:
+                continue
             self.register_size_names[(reg.vex_offset, reg.size)] = reg.name
             for name, off, sz in reg.subregisters:
                 # special hacks for X86 and AMD64 - don't translate register names to bp, sp, etc.
