--- a/w3af_console
+++ b/w3af_console
@@ -34,14 +34,6 @@ Options:
     -s <script_file> or --script=<script_file>
         Run <script_file> script
 
-    -n or --no-update
-        No update check will be made when starting. This option takes 
-        precedence over the 'auto-update' setting in 'startup.conf' file.
-     
-    -f or --force-update
-        An update check will be made when starting. This option takes 
-        precedence over the 'auto-update' setting in 'startup.conf' file.
-     
     -p <profile> or --profile=<profile>
         Run with the selected <profile>
     
@@ -110,8 +102,8 @@ def _generate_run_commands(script_file,
 
 def main():
     try:
-        long_options = ['script=', 'help', 'version', 'no-update',
-                        'force-update', 'profile=', 'commit=', 'profile-run']
+        long_options = ['script=', 'help', 'version',
+                        'profile=', 'profile-run']
         opts, _ = getopt.getopt(sys.argv[1:], "ehvs:nfp:P:", long_options)
     except getopt.GetoptError:
         # Print help information and exit
@@ -121,7 +113,7 @@ def main():
     script_file = None
     force_profile = None
     profile = None
-    do_update = None
+    do_update = False
     
     for o, a in opts:
         if o == "-e":
@@ -144,10 +136,6 @@ def main():
         if o in ('-v', '--version'):
             print(get_w3af_version())
             return 0
-        if o in ('-f', '--force-update'):
-            do_update = True
-        elif o in ('-n', '--no-update'):
-            do_update = False
     
     commands_to_run = _generate_run_commands(script_file, profile, force_profile)
     console = ConsoleUI(commands=commands_to_run, do_upd=do_update)
--- a/w3af_gui
+++ b/w3af_gui
@@ -27,14 +27,6 @@ Options:
     -h or --help
         Display this help message.
 
-    -n or --no-update
-        No update check will be made when starting. This option takes 
-        precedence over the 'auto-update' setting in 'startup.conf' file.
-     
-    -f or --force-update
-        An update check will be made when starting. This option takes 
-        precedence over the 'auto-update' setting in 'startup.conf' file.
-    
     -p <profile> or --profile=<profile>
         Run with the selected <profile>
 
@@ -61,7 +53,7 @@ def usage():
 
 def main():
     try:
-        long_options = ['help', 'no-update', 'force-update', 'profile=']
+        long_options = ['help', 'profile=']
         opts, _ = getopt.getopt(sys.argv[1:], "ehnfp:", long_options)
     except getopt.GetoptError:
         # print help information and exit:
@@ -69,7 +61,7 @@ def main():
         return -3
     
     profile = None
-    doupdate = None
+    doupdate = False
     
     for o, a in opts:
         if o in ("-e"):
@@ -83,11 +75,6 @@ def main():
         if o in ('-h', '--help'):
             usage()
             return 0
-        if o in ('-f', '--force-update'):
-            doupdate = True
-        elif o in ('-n', '--no-update'):
-            doupdate = False
-    
 
     # go with GTK, but first check about DISPLAY environment variable
     if sys.platform != "win32":

