diff --git a/roguehostapd/config/hostapdconfig.py b/roguehostapd/config/hostapdconfig.py
index dcb7422..0dd4aee 100644
--- a/roguehostapd/config/hostapdconfig.py
+++ b/roguehostapd/config/hostapdconfig.py
@@ -6,7 +6,7 @@ import re
 import collections
 import os
 import json
-from ConfigParser import SafeConfigParser
+from configparser import SafeConfigParser
 
 def get_default_settings():
     """
