-- setup.py.orig 2016-08-31 20:26:43.831412367 -0400
+++ setup.py 2016-08-31 20:27:11.407699599 -0400
@@ -37,6 +37,8 @@
 
         spawn(['tar', '-zxf', self.NIDSTAR], search_path = 1)
         os.chdir(self.NIDSDIR)
+        spawn([pathjoin('/usr/bin/','curl'), 'http://git.savannah.gnu.org/cgit/config.git/plain/config.guess', '-O'])
+        spawn([pathjoin('/usr/bin/','curl'), 'http://git.savannah.gnu.org/cgit/config.git/plain/config.sub', '-O'])
         spawn([pathjoin('.','configure'), 'CFLAGS=-fPIC'])
         spawn(['make'], search_path = 1)
         os.chdir('..')
