diff -aur pip3line-2.7.1.orig/defaultplugins/baseplugins/CMakeLists.txt pip3line-2.7.1/defaultplugins/baseplugins/CMakeLists.txt
--- pip3line-2.7.1.orig/defaultplugins/baseplugins/CMakeLists.txt	2016-05-09 21:24:10.000898016 -0400
+++ pip3line-2.7.1/defaultplugins/baseplugins/CMakeLists.txt	2016-05-09 21:24:31.683909522 -0400
@@ -278,7 +278,7 @@
 target_link_libraries ( baseplugins ${EXTERN_LIB} )
 
 IF(UNIX)
-	install (TARGETS baseplugins DESTINATION share/pip3line/
+	install (TARGETS baseplugins DESTINATION lib/pip3line/
 		PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
 		GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 ENDIF(UNIX)
diff -aur pip3line-2.7.1.orig/defaultplugins/distormplugin/CMakeLists.txt pip3line-2.7.1/defaultplugins/distormplugin/CMakeLists.txt
--- pip3line-2.7.1.orig/defaultplugins/distormplugin/CMakeLists.txt	2016-05-09 21:24:10.010897867 -0400
+++ pip3line-2.7.1/defaultplugins/distormplugin/CMakeLists.txt	2016-05-09 21:24:49.803640589 -0400
@@ -88,7 +88,7 @@
 target_link_libraries ( distormplugin  ${EXTERN_LIBS} )
 
 IF(UNIX)
-	install (TARGETS distormplugin DESTINATION share/pip3line/
+	install (TARGETS distormplugin DESTINATION lib/pip3line/
 		PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
 		GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE )
 ENDIF(UNIX)
diff -aur pip3line-2.7.1.orig/defaultplugins/exampleplugin/CMakeLists.txt pip3line-2.7.1/defaultplugins/exampleplugin/CMakeLists.txt
--- pip3line-2.7.1.orig/defaultplugins/exampleplugin/CMakeLists.txt	2016-05-09 21:24:10.010897867 -0400
+++ pip3line-2.7.1/defaultplugins/exampleplugin/CMakeLists.txt	2016-05-09 21:25:10.370002009 -0400
@@ -59,7 +59,7 @@
 target_link_libraries ( exampleplugin ${EXTERN_LIB} )
 
 IF(UNIX)
-	install (TARGETS exampleplugin DESTINATION share/pip3line/
+	install (TARGETS exampleplugin DESTINATION lib/pip3line/
 		PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
 		GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 ENDIF(UNIX)
diff -aur pip3line-2.7.1.orig/defaultplugins/opensslplugin/CMakeLists.txt pip3line-2.7.1/defaultplugins/opensslplugin/CMakeLists.txt
--- pip3line-2.7.1.orig/defaultplugins/opensslplugin/CMakeLists.txt	2016-05-09 21:24:10.010897867 -0400
+++ pip3line-2.7.1/defaultplugins/opensslplugin/CMakeLists.txt	2016-05-09 21:25:31.009695680 -0400
@@ -88,7 +88,7 @@
 	target_link_libraries ( opensslplugin  ${EXTERN_LIBS} )
 
 	IF(UNIX)
-		install (TARGETS opensslplugin DESTINATION share/pip3line/
+		install (TARGETS opensslplugin DESTINATION lib/pip3line/
 			PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
 			GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE )
 	ENDIF(UNIX)
diff -aur pip3line-2.7.1.orig/defaultplugins/python27plugin/CMakeLists.txt pip3line-2.7.1/defaultplugins/python27plugin/CMakeLists.txt
--- pip3line-2.7.1.orig/defaultplugins/python27plugin/CMakeLists.txt	2016-05-09 21:24:10.010897867 -0400
+++ pip3line-2.7.1/defaultplugins/python27plugin/CMakeLists.txt	2016-05-09 21:26:01.809238547 -0400
@@ -113,7 +113,7 @@
 
 
 IF(UNIX)
-	install (TARGETS python27plugin DESTINATION share/pip3line/
+	install (TARGETS python27plugin DESTINATION lib/pip3line/
 		PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
 		GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 ENDIF(UNIX)
diff -aur pip3line-2.7.1.orig/defaultplugins/pythonplugin/CMakeLists.txt pip3line-2.7.1/defaultplugins/pythonplugin/CMakeLists.txt
--- pip3line-2.7.1.orig/defaultplugins/pythonplugin/CMakeLists.txt	2016-05-09 21:24:10.010897867 -0400
+++ pip3line-2.7.1/defaultplugins/pythonplugin/CMakeLists.txt	2016-05-09 21:26:22.122270394 -0400
@@ -96,7 +96,7 @@
 	target_link_libraries ( python3plugin ${EXTERN_LIB} ${PYTHON_LIBRARIES} )
 
 	IF(UNIX)
-		install (TARGETS python3plugin DESTINATION share/pip3line/
+		install (TARGETS python3plugin DESTINATION lib/pip3line/
 			PERMISSIONS OWNER_READ OWNER_WRITE OWNER_EXECUTE
 			GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 	ENDIF(UNIX)
