--- Makefile.orig	2015-11-09 20:07:25.279288991 -0500
+++ Makefile	2015-11-09 20:07:38.659087575 -0500
@@ -13,9 +13,8 @@
 MANDIR=/usr/share/man/man8
 SRCDIR=.
 CC=gcc
-USE_SSE=-msse -mfpmath=sse
 STRIP=-s
-CFLAGS=-W -Wall -Wextra -O3 $(USE_SSE) -ffast-math $(STRIP)
+CFLAGS=-W -Wall -Wextra -O3 -ffast-math $(STRIP)
 INCLUDES=-I$(SRCDIR)/include
 DFLAGS=-D__HAVE_TURBO__ -DVERSION=\"$(shell cat ../VERSION)\"
 #DFLAGS+=-D__HAVE_DEBUG__
