diff -aur zzuf-0.15.orig/src/libzzuf/lib-mem.c zzuf-0.15/src/libzzuf/lib-mem.c
--- zzuf-0.15.orig/src/libzzuf/lib-mem.c	2016-05-03 21:33:32.784451451 -0400
+++ zzuf-0.15/src/libzzuf/lib-mem.c	2016-05-03 21:36:07.699555883 -0400
@@ -20,6 +20,7 @@
 #define _GNU_SOURCE
 /* Need this for MAP_ANON and valloc() on FreeBSD (together with cdefs.h) */
 #define _BSD_SOURCE
+#define _DEFAULT_SOURCE
 #if defined HAVE_SYS_CDEFS_H
 #   include <sys/cdefs.h>
 #endif
diff -aur zzuf-0.15.orig/src/myfork.c zzuf-0.15/src/myfork.c
--- zzuf-0.15.orig/src/myfork.c	2016-05-03 21:33:32.781118152 -0400
+++ zzuf-0.15/src/myfork.c	2016-05-03 21:36:31.912647386 -0400
@@ -19,6 +19,7 @@
 
 #define _INCLUDE_POSIX_SOURCE /* for STDERR_FILENO on HP-UX */
 #define _BSD_SOURCE /* for setenv on glibc systems */
+#define _DEFAULT_SOURCE
 
 #if defined HAVE_STDINT_H
 #   include <stdint.h>
diff -aur zzuf-0.15.orig/src/zzuf.c zzuf-0.15/src/zzuf.c
--- zzuf-0.15.orig/src/zzuf.c	2016-05-03 21:33:32.784451451 -0400
+++ zzuf-0.15/src/zzuf.c	2016-05-03 21:36:51.249121328 -0400
@@ -20,6 +20,7 @@
 #define _INCLUDE_POSIX_SOURCE /* for STDERR_FILENO on HP-UX */
 #define _POSIX_SOURCE /* for kill() on glibc systems */
 #define _BSD_SOURCE /* for setenv() on glibc systems */
+#define _DEFAULT_SOURCE
 
 #if defined HAVE_STDINT_H
 #   include <stdint.h>
diff -aur zzuf-0.15.orig/test/bug-mmap.c zzuf-0.15/test/bug-mmap.c
--- zzuf-0.15.orig/test/bug-mmap.c	2016-05-03 21:33:32.787784749 -0400
+++ zzuf-0.15/test/bug-mmap.c	2016-05-03 21:37:17.012198414 -0400
@@ -14,6 +14,7 @@
 #include "config.h"
 
 #define _BSD_SOURCE 1 /* for MAP_POPULATE */
+#define _DEFAULT_SOURCE
 
 #if HAVE_SYS_MMAN_H
 #   include <sys/mman.h>
