diff -aur r2pipe-0.8.4.orig/r2pipe/__init__.py r2pipe-0.8.4/r2pipe/__init__.py
--- r2pipe-0.8.4.orig/r2pipe/__init__.py	2016-06-05 20:18:10.516058922 -0400
+++ r2pipe-0.8.4/r2pipe/__init__.py	2016-06-05 20:19:17.311768465 -0400
@@ -99,7 +99,7 @@
 			Returns an object with methods to interact with r2 via commands
 		"""
 		if in_rlang():
-			print "RLANG IS SET"
+			print ("RLANG IS SET")
 			self._cmd = self._cmd_rlang
 			return
 		self._cmd = self._cmd_rlang
