Only in nfqueue-bindings-nfqueue-bindings-0.5: build
diff -c nfqueue-bindings-nfqueue-bindings-0.5/CMakeLists.txt nfqueue-bindings-update/CMakeLists.txt
*** nfqueue-bindings-nfqueue-bindings-0.5/CMakeLists.txt        2014-07-29 04:25:31.000000000 -0400
--- nfqueue-bindings-update/CMakeLists.txt      2016-03-14 12:09:09.729759073 -0400
***************
*** 30,36 ****
  CHECK_FUNCTION_EXISTS(nfq_set_verdict2 HAVE_NFQ_SET_VERDICT2)
  
  ADD_SUBDIRECTORY(python)
- ADD_SUBDIRECTORY(perl)
  
  CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/config.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/config.h)
  CONFIGURE_FILE(${CMAKE_CURRENT_SOURCE_DIR}/nfq_version.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/nfq_version.h)
--- 30,35 ----
Common subdirectories: nfqueue-bindings-nfqueue-bindings-0.5/examples and nfqueue-bindings-update/examples
diff -c nfqueue-bindings-nfqueue-bindings-0.5/Makefile nfqueue-bindings-update/Makefile
*** nfqueue-bindings-nfqueue-bindings-0.5/Makefile      2014-07-29 04:25:31.000000000 -0400
--- nfqueue-bindings-update/Makefile    2016-03-14 12:10:01.299761679 -0400
***************
*** 1,5 ****
  PREFIX = /usr/local
! CMAKE_OPTIONS = -DDEBUG=1 -DCMAKE_VERBOSE_MAKEFILE=0 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=$(PREFIX)
  CMAKE_EXTRA_OPTIONS =
  BUILD_DIR = build
  
--- 1,6 ----
  PREFIX = /usr/local
! PYTHON_EXE=`which python2`
! CMAKE_OPTIONS = -DDEBUG=1 -DCMAKE_VERBOSE_MAKEFILE=0 -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=$(PREFIX)  -DPYTHON_INCLUDE_DIR=/usr/include/python2.7 -DPYTHON_LIBRARY=/usr/lib/libpython2.7.so  -DPYTHON_EXECUTABLE=$(PYTHON_EXE)
  CMAKE_EXTRA_OPTIONS =
  BUILD_DIR = build
  
Common subdirectories: nfqueue-bindings-nfqueue-bindings-0.5/perl and nfqueue-bindings-update/perl
Common subdirectories: nfqueue-bindings-nfqueue-bindings-0.5/python and nfqueue-bindings-update/python
