--- Makefile.am	2012-07-17 08:53:51.000000000 +0200
+++ Makefile.am	2012-11-20 21:56:11.000000000 +0100
@@ -30,8 +30,8 @@
 	$(INSTALL) -d $(DESTDIR)$(localstatedir)/opendnssec/signconf
 	$(INSTALL) -d $(DESTDIR)$(localstatedir)/opendnssec/unsigned
 	$(INSTALL) -d $(DESTDIR)$(localstatedir)/opendnssec/signed
-	$(INSTALL) -d $(DESTDIR)$(localstatedir)/run
-	$(INSTALL) -d $(DESTDIR)$(localstatedir)/run/opendnssec
+	$(INSTALL) -d $(DESTDIR)/var/run
+	$(INSTALL) -d $(DESTDIR)/var/run/opendnssec
 
 docs:
 	(cd libhsm; $(MAKE) doxygen)
--- m4/opendnssec_common.m4	2012-11-01 11:56:59.000000000 +0100
+++ m4/opendnssec_common.m4	2012-11-20 21:52:56.000000000 +0100
@@ -19,7 +19,7 @@
 OPENDNSSEC_DATA_DIR=$full_datadir/opendnssec
 OPENDNSSEC_SYSCONF_DIR=$full_sysconfdir/opendnssec
 OPENDNSSEC_LOCALSTATE_DIR="$full_localstatedir/opendnssec"
-OPENDNSSEC_PID_DIR="$full_localstatedir/run/opendnssec"
+OPENDNSSEC_PID_DIR="/var/run/opendnssec"
 
 AC_SUBST([OPENDNSSEC_BIN_DIR])
 AC_SUBST([OPENDNSSEC_SBIN_DIR])
