diff -aur intrace-1.5.orig/config.h intrace-1.5/config.h
--- intrace-1.5.orig/config.h	2016-05-29 19:56:50.280203363 -0400
+++ intrace-1.5/config.h	2016-05-29 19:59:39.764447998 -0400
@@ -16,6 +16,7 @@
 /* struct tcphdr incompabilities */
 #ifndef _BSD_SOURCE
 #define _BSD_SOURCE
+#define _DEFAULT_SOURCE
 #endif
 
 #define MAX_HOPS 32
diff -aur intrace-1.5.orig/debug.h intrace-1.5/debug.h
--- intrace-1.5.orig/debug.h	2016-05-29 19:56:50.280203363 -0400
+++ intrace-1.5/debug.h	2016-05-29 19:59:15.291464397 -0400
@@ -17,7 +17,7 @@
 extern int _debug_printf(debug_level_t dl, const char *file, const char *func, int line, const char *fmt, ...);
 
 /* Better wrapper */
-#define debug_printf(dl, ...) _debug_printf(dl, __FILE__, __FUNCTION__, __LINE__, __VA_ARGS__);
+#define debug_printf(dl, ...) _debug_printf(dl, __FILE__, __func__, __LINE__, __VA_ARGS__);
 
 /* Function initializes debug */
 extern int _debug_init(debug_level_t dl, char *l);
